runtime/minor_gc.c

```diff
@@ -129,11 +131,10 @@ void caml_set_minor_heap_size (asize_t wsize)
 {
   caml_domain_state* domain_state = Caml_state;
   struct caml_minor_tables *r = domain_state->minor_tables;
-  if (domain_state->young_ptr != domain_state->young_end) caml_minor_collection ();
 
-  if(caml_reallocate_minor_heap(wsize) < 0) {
-    caml_fatal_error("Fatal error: No memory for minor heap");
-  }
+  global_minor_heap_wsz_per_domain = wsize;
+
+  if (domain_state->young_ptr != domain_state->young_end) caml_minor_collection ();
```

Comment:
I think we should always trigger a minor collection here, there are conditions where a minor collection won't be hit on next gc poll